src/plugins/discover/public/application/embeddable/saved_search_embeddable.tsx

```diff
@@ -344,64 +191,226 @@ export class SearchEmbeddable
-    searchScope.columns = handleSourceColumnState(
+    searchProps.columns = handleSourceColumnState(
       { columns: this.input.columns || this.savedSearch.columns },
       this.services.core.uiSettings
     ).columns;
+
     const savedSearchSort =
       this.savedSearch.sort && this.savedSearch.sort.length
         ? this.savedSearch.sort
         : getDefaultSort(
-            this.searchScope?.indexPattern,
+            this.searchProps?.indexPattern,
             getServices().uiSettings.get(SORT_DEFAULT_ORDER_SETTING, 'desc')
           );
-    searchScope.sort = this.input.sort || savedSearchSort;
-    searchScope.sharedItemTitle = this.panelTitle;
-
+    searchProps.sort = this.input.sort || savedSearchSort;
+    searchProps.sharedItemTitle = this.panelTitle;
     if (forceFetch || isFetchRequired) {
-      this.filtersSearchSource!.setField('filter', this.input.filters);
-      this.filtersSearchSource!.setField('query', this.input.query);
+      this.filtersSearchSource.setField('filter', this.input.filters);
+      this.filtersSearchSource.setField('query', this.input.query);
       if (this.input.query?.query || this.input.filters?.length) {
-        this.filtersSearchSource!.setField('highlightAll', true);
+        this.filtersSearchSource.setField('highlightAll', true);
       } else {
-        this.filtersSearchSource!.removeField('highlightAll');
+        this.filtersSearchSource.removeField('highlightAll');
       }
 
       this.prevFilters = this.input.filters;
       this.prevQuery = this.input.query;
       this.prevTimeRange = this.input.timeRange;
       this.prevSearchSessionId = this.input.searchSessionId;
-      this.fetch();
-    } else if (this.searchScope) {
-      // trigger a digest cycle to make sure non-fetch relevant changes are propagated
-      this.searchScope.$applyAsync();
+      this.searchProps = searchProps;
+      await this.fetch();
+      if (this.node) {
+        await this.rerenderComponent(this.node);
+      }
+    } else if (this.searchProps && this.node) {
+      this.searchProps = searchProps;
+      await this.renderReactComponent(this.node, this.searchProps);
+    }
+  }
+
+  /**
+   *
+   * @param {Element} domNode
+   */
+  public async render(domNode: HTMLElement) {
+    if (!this.searchProps) {
+      throw new Error('Search props not defined');
+    }
+    if (this.node) {
+      ReactDOM.unmountComponentAtNode(this.node);
     }
+    this.node = domNode;
+  }
+
+  private async rerenderComponent(domNode: HTMLElement) {
+    if (!this.searchProps) {
+      return;
+    }
+    this.searchProps.refs = domNode;
+    await this.pushContainerStateParamsToProps(this.searchProps);
+    await this.renderReactComponent(domNode, this.searchProps);
+  }
+
+  private async renderReactComponent(domNode: HTMLElement, searchProps: SearchProps) {
+    return new Promise(() => {
+      ReactDOM.render(<SavedSearchEmbeddableComponentMemoized {...searchProps} />, domNode);
+    });
```

Comment:
:question: Could you clarify for me, why we create a new promise here? Are we trying to "delay" the rendering to the next execution slot for some reason? (Which would not work via that code, since the executor of a Promise is executed synchronously.) Because the way this is currently written we return a promise that will never resolve, and thus potentially creating a memory leak, since we're waiting on it in other places with `await` (which will never pass).

Reply:
Matthias says it makes debugging easier :)

Reply:
but there's something missing in the code? one moment

Reply:
The 3rd param of `ReactDom.render` will be executed once rendering has finished, it should resolve the Promise

Reply:
However fine with doing it simpler, main purpose was just to use 1` ReactDOM.render` instead of 2

Reply:
Ah okay, I see, you planned to do that as:

```ts
```

Reply:
If that actually helps lowering the amount of rendering, it might be a reasonable change. Though we should keep in mind, that the 3rd parameter or `render` will be called on every Render or update, meaning React will also keep a reference around to that to call it again when updated, so I think we should make sure we're not accidentally leaking promises around between re-renderings (that anyway can only resolve once).

Reply:
did that because I first thought this might be the issue for the dom node count increase, it wasn't. fine with keeping it the way it was used before .. for now